docs/src/howto/contribute.md

```diff
@@ -1 +1,140 @@
-# Contribute 
+# How to contribute to Canonical Kubernetes
+
+Canonical Kubernetes is proudly open source, published under the GPLv3 license.
+We welcome and encourage contributions to the code and the documentation.
+
+## Contribute to the code
+
+Canonical Kubernetes is shipped as a snap package. To contribute to the code,
+you should first make sure you can build and test the snap locally.
+
+### Build the snap
+
+To build the snap locally, you will need the following:
+
+- The latest, snap-based version of LXD (see the [install guide here][install
+   lxd])
+- The Snapcraft build tool, for building the snap (see the [Snapcraft
+   documentation][]).
+
+Clone the [GitHub repository for the k8s snap][code repo] and then open a terminal in that
+directory. Run the command:
+
+```
+snapcraft --use-lxd
+```
+
+This will launch an LXD container and use it to build a version of the snap.
+This will take some time as the build process fetches dependencies, stages the
+‘parts’ of the snap and creates the snap package itself. The snap itself will
+be fetched from the build environment and placed in the local project
+directory. Note that the LXD container used for building will be stopped, but
+not deleted. This is in case there were any errors or artefacts you may wish to
+inspect.
+
+### Install the snap
+
+The snap can then be installed locally by using the ‘–dangerous’ option. This
+is a safeguard to make sure the user is aware that the snap is not signed by
+the snap store, and is not confined:
+
+```
+sudo snap install k8s_v1.29.2_multi.snap --dangerous --classic
+```
+
+```{note} You will not be able to install this snap if there is already a k8s snap installed on your system.
+```
+
+Once you have verified the current snap build works, it can be removed with:
+
+```
+sudo snap remove k8s --purge
+```
+
+The `purge` option is recommended when iterating over code changes, as it also
+removes all the installed artefacts which may be associated with the snap.
+
+Now you can iterate over changes to the snap, rebuild and test.
+
+As noted previously, the LXD container used for building is not removed and
+will be reused by subsequent build instructions. When you are satisfied it is
+no longer needed, this container can be removed:
+
+```
+lxc delete snapcraft-microk8s
+```
+
+### Contribute changes
+
+We welcome any improvements and bug-fixes to the Canonical Kubernetes code.
+Once you have tested your changes, please make a pull request on the [code
+repository][code repo] and we will review it as soon as possible.
+
+
+## Contribute to the documentation
+
+Our aim is to provide easy-to-understand documentation on all aspects of
+Canonical Kubernetes, so we greatly appreciate your feedback and contributions.
```

Comment:
We could include a reference here to where we can be reached Discourse, Slack, Mattermost etc. This could also serve to provide feedback that's not a gh issue or PR.

Reply:
yes, good point. I will just put a link to the community page where all those things are mentioned

Reply:
I added a link but for some reason it didn't sync. Will put it in the next update. Thanks!